### PR TITLE
Handle interrupts in prow controllers

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/5074

### CODE REVIEW

File: prow/cmd/plank/main.go [hunk 1]
```diff
@@ -94,6 +100,13 @@ func main() {
 		logrus.WithError(err).Fatal("Error creating plank controller.")
 	}
 	for range time.Tick(30 * time.Second) {
+		select {
+		case <-sig:
+			logrus.Infof("plank is shutting down...")
```
Comment (cjwagner) [inline prow/cmd/plank/main.go]: Capitalize "Plank".

File: prow/cmd/jenkins-operator/main.go [hunk 2]
```diff
@@ -99,6 +102,9 @@ func main() {
 		ghc = github.NewClient(oauthSecret, *githubEndpoint)
 	}
 
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, os.Kill, syscall.SIGTERM)
```
Comment (cjwagner) [inline prow/cmd/jenkins-operator/main.go]: The kill signal cannot be received by a process, the kernel just kills the process so you don't need to include it here or in plank.

File: prow/cmd/jenkins-operator/main.go [hunk 3]
```diff
@@ -111,6 +117,13 @@ func main() {
 	go serveLogs(jc)
 
 	for range time.Tick(30 * time.Second) {
+		select {
```
Comment (spxtr) [inline prow/cmd/jenkins-operator/main.go]: Can you combine this and the `for range time.Tick(...)` into one `select`?

File: prow/cmd/jenkins-operator/main.go (hunk 2 above)
Comment (kargakis) [reply prow/cmd/jenkins-operator/main.go]: ack

File: prow/cmd/plank/main.go (hunk 1 above)
Comment (kargakis) [reply prow/cmd/plank/main.go]: ack

File: prow/cmd/jenkins-operator/main.go (hunk 3 above)
Comment (kargakis) [reply prow/cmd/jenkins-operator/main.go]: ack

**Pull request merged**